jerry-core/parser/js/js-lexer.c

```diff
@@ -1429,11 +1429,36 @@ lexer_parse_number (parser_context_t *context_p) /**< context */
       }
 
       lexer_check_numbers (context_p, &source_p, source_end_p, LIT_CHAR_7, true);
+
+      if (source_p[0] >= LIT_CHAR_8
+          && source_p[0] <= LIT_CHAR_9)
+      {
+#if ENABLED (JERRY_ESNEXT)
+        lexer_check_numbers (context_p, &source_p, source_end_p, LIT_CHAR_9, false);
+        context_p->token.extra_value = LEXER_NUMBER_DECIMAL;
+#else
+        parser_raise_error (context_p, PARSER_ERR_INVALID_NUMBER);
+#endif /* ENABLED (JERRY_ESNEXT) */
+      }
```

Comment:
Is it possible to combine the two cases somehow to reduce code size?

Reply:
There was no reply to this question. The `else` in line 1444 seems es5.1 specific, and all esnext cases could be merged here.

Reply:
I am sorry for not answering to your question. I Simplified the code.